src/warc2zim/content_rewriting/ds.py

```diff
@@ -256,14 +255,15 @@ def filter_by_bitrate(array, max_bitrate, mime):
 ]
 
 
-def build_domain_specific_rewriter(path: str, url_rewriter):
+def get_ds_rules(path: str):
     """
-    This build a JsRewriter with extra rules depending of the path.
+    This build a domain specifc Rewriter (given in rewriter_class), by passing a set of
+    extra rules to the rewriter if needed.
     """
 
     for rule in RULES:
-        for contain in rule.contains:
-            if contain in path:
-                return JsRewriter(url_rewriter, rule.rx_rules)
+        for pattern in rule.contains:
+            if pattern in path:
+                return rule.rx_rules
```

Comment:
This excludes matching multiple rules. Is this on purpose? I can't think of any example of a single URL matching several rules but then maybe we should add a comment on top of `RULES` to indicate that rules must be sorted by higher precision first (although I understand those are simply copied from upstream here)